### PR TITLE
Added GLB code to ClassVal Checker, Standard GLB isn't correct for th…

### PR DESCRIPTION
**Pull request by smillst**
…is checker

### CODE REVIEW

**Comment by mernst** [comment]
Can you add tests for this code?

File: framework/tests/classval/GLBTest.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    // If that code changes, this code may not test GLB anymore
+    // This code does not test correctness. Because no experssion is given the GLB as a type,
+    // it is impossible to test GLB for correctness.
+    //:: error: (type.argument.type.incompatible) :: error: (assignment.type.incompatible)
+    GLBTest<@ClassVal({"A","B","C"}) ?> f1 = new GLBTest<@ClassVal({"A","E"}) Object>();
+    //:: error: (type.argument.type.incompatible) :: error: (assignment.type.incompatible)
+    GLBTest<@ClassVal({"A","B","C"}) ?> f2 = new GLBTest<@ClassBound({"A","E"}) Object>();
+    //:: error: (type.argument.type.incompatible) :: error: (assignment.type.incompatible)
+    GLBTest<@ClassBound({"A","B","C"}) ?> f3 = new GLBTest<@ClassBound({"A","E"}) Object>();
+}
```
Comment (wmdietl) [inline framework/tests/classval/GLBTest.java]: Can you also add a few passing cases?

File: framework/tests/classval/GLBTest.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import org.checkerframework.common.reflection.qual.ClassBound;
+import org.checkerframework.common.reflection.qual.ClassVal;
+import org.checkerframework.common.reflection.qual.ClassValBottom;
+
+class GLBTest<@ClassVal({"A","B"}) T extends Object> {
+    // This code is intented to cover the more complex branchs for the GLB calcuation.
+    // This only triggers the GLB calculation because of a hack in
+    // org.checkerframework.framework.util.AnnotatedTypes.addAnnotationsImpl()
+    // If that code changes, this code may not test GLB anymore
```
Comment (wmdietl) [inline framework/tests/classval/GLBTest.java]: Periods at ends of sentences.

File: framework/tests/classval/GLBTest.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    // This code does not test correctness. Because no experssion is given the GLB as a type,
```
Comment (wmdietl) [inline framework/tests/classval/GLBTest.java]: experssion -> expression

**Pull request merged**